pkg/collector/corechecks/net/ntp.go

```diff
@@ -76,7 +77,7 @@ func (c *ntpConfig) parse(data []byte, initData []byte, getLocalServers func() (
 	defaultPort := 123
 	defaultOffsetThreshold := 60
 
-	defaultHosts := util.GetCloudProviderNTPHosts()
+	defaultHosts := util.GetCloudProviderNTPHosts(context.TODO())
```

Comment:
What's the plan for removing this TODO?

Reply:
This `TODO()` is in a core check.
Unfortunately, today, there is no timeout for checks execution.
There are some checks which are known to last several minutes like the KSM or SNMP ones for ex, but we cannot know for the custom checks written by our users.
Introducing a time-out for check execution would be a breaking change that is not in the scope of this work.

As a consequence, we could imagine to remove this `TODO()` and to add a `context.Context` parameter to this function in order to move the `context.Context` creation upper in the function call tree.
But anyhow, the `context.Context` created in the scope of a (core) check will be an “empty” context (either `context.Background()` or `context.TODO()`, so, it wouldn’t change the behaviour.

The code could look better if all the `context.Context` creation were moved as high as possible in the function call tree but I though I would do it in another PR because this one already implements the `context.Context` propagation in the scopes where a deadline can be defined (i.e.: event loops monitored by health probe, but not in checks, nor in interactive agent CLI commands)

Reply:
Thanks for the explanation!

> The code could look better if all the context.Context creation were moved as high as possible in the function call tree but I though I would do it in another PR because this one already implements the context.Context propagation in the scopes where a deadline can be defined

that makes sense, let's do it in another PR

Also, it would be good IMO to track `TODO`s somewhere (at least for the ones that are not going to be just replaced by `Background`)